sdk/src/test/java/io/dapr/it/DaprIntegrationTestingRunner.java

```diff
@@ -0,0 +1,87 @@
+package io.dapr.it;
+
+import org.junit.Assert;
+
+import java.io.*;
+import java.util.UUID;
+import java.util.concurrent.*;
+
+
+public class DaprIntegrationTestingRunner {
+
+    private Runtime rt = Runtime.getRuntime();
+    private Process proc;
+
+    private  String successMessage;
+    private Class serviceClass;
+    private Boolean isGrpc;
+    private int port;
+    private int sleepTime;
+    private String appName;
+
+    private DaprIntegrationTestingRunner(String successMessage, Class serviceClass, Boolean isGrpc, int port, int sleepTime) {
+        this.successMessage = successMessage;
+        this.serviceClass = serviceClass;
+        this.isGrpc = isGrpc;
+        this.port = port;
+        this.sleepTime = sleepTime;
+        this.generateAppName();
+    }
+
+    public static DaprIntegrationTestingRunner createDaprIntegrationTestingRunner(String successMessage, Class serviceClass, Boolean isGrpc, int port, int sleepTime) {
+        return new DaprIntegrationTestingRunner(successMessage, serviceClass, isGrpc, port, sleepTime);
+    }
+
+
+    public void initializeDapr() throws IOException, InterruptedException, TimeoutException, ExecutionException {
+
+        File file= new File("../");
+        proc= rt.exec(this.buildDaprCommand(), null,file);
+        InputStream stdin = proc.getInputStream();
+        InputStreamReader isr = new InputStreamReader(stdin);
+        BufferedReader br = new BufferedReader(isr);
+
+        final Runnable stuffToDo = new Thread(() -> {
+            try {
+                String line ;
+                while ((line = br.readLine()) != null) {
+                    System.out.println(line);
+                    if (line.contains(successMessage)) {
+                        break;
+                    }
+                }
+            }catch(IOException ex){
+                Assert.fail(ex.getMessage());
+            }
+        });
+
+        final ExecutorService executor = Executors.newSingleThreadExecutor();
+        final Future future = executor.submit(stuffToDo);
+        executor.shutdown(); // This does not cancel the already-scheduled task.
+        future.get(2, TimeUnit.MINUTES);
+        Thread.sleep(sleepTime);
+
+    }
+
+    private static final String DAPR_RUN = "dapr run --app-id %s ";
+    private static final String DAPR_COMMAND = " -- mvn exec:java -pl=sdk -D exec.mainClass=%s -D exec.classpathScope=\"test\"";
```

Comment:
You don't actually need to start the app here. You can just start dapr without the service and then start the test app directly in the same JVM (separate thread).

Make DAPR_COMMAND = " -- cat" to just block.
Start a new thread, starting the service: io.dapr.e2e.apps.MyService.main()

Reply:
I am getting the following error with the cat command, even when that work on my terminal
![image](https://user-images.githubusercontent.com/35985447/72121750-8ebe6080-3321-11ea-96b3-b918c2baba1e.png)

But using DAPR on windows 10, I am getting the following:
exec: "cat": executable file not found in %!P(MISSING)ATH%!(NOVERB)

Reply:
Technically it can be any blocking command.